components/PlantList.tsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { Text, View } from "./Themed";
+import { FlatList, StyleSheet, Image, TouchableOpacity } from "react-native";
+import { mockData } from "../services/mockData";
+import { FontAwesome5 } from "@expo/vector-icons";
+import { NavigationProp } from "@react-navigation/native";
+
+type Props = {
+  navigation: NavigationProp<any>; // ! get typed properly
+};
+
+export const PlantList = ({ navigation }: Props) => {
+  const { plants } = mockData;
+
+  return plants ? (
+    <FlatList
+      style={styles.container}
+      data={plants}
+      renderItem={({ item }) => (
+        <TouchableOpacity
+          style={[styles.plantContainer]}
+          onPress={() =>
+            navigation.navigate("PlantInfoScreen", {
+              title: item.name,
+              plantInfo: item,
+            })
+          }
+        >
+          <View key={item.id} style={styles.plantL}>
```

Comment:
Could be worth considering using

keyExtractor={(item) => item.id}

as a Flatlist property, as it would be more nativey and consisitent with other flatlist components in app.

No biggie I just wanted to feel like a pro and add a comment.

Reply:
Good spot I had no idea about keyExtractor and just assumed it was like react. It looks like we might not even need keyExtractor as long as your item has a key or id property. Tough going off some posts it looks like you still get a console error about using a key if you don't supply keyExtractor even though its picked it up automatically. We'll just keep using it to avoid the error and hopefully spot it quicker when we actually forget to have an id with an item.

https://reactnative.dev/docs/flatlist#keyextractor
Used to extract a unique key for a given item at the specified index. Key is used for caching and as the react key to track item re-ordering. <ins>The default extractor checks item.key, then item.id, and then falls back to using the index, like React does.</ins>

https://blog.logrocket.com/deep-dive-react-native-flatlist/
_The keyExtractor prop will retrieve a unique key for each item in the array
N.B., if your array contains a key or id field, you don’t need to include this prop. By default, FlatList will look for the key or id property._